it/uk/gov/hmrc/agentassurance/controllers/AgentAssuranceControllerISpec.scala

```diff
@@ -370,4 +377,87 @@ class AgentAssuranceControllerISpec extends IntegrationSpec
       response.status shouldBe 401
     }
   }
+
+  feature("/amls/create") {
+
+    val amlsCreateUrl = s"http://localhost:$port/agent-assurance/amls"
+
+    def doRequest(payload: String) =
+      Await.result(
+        wsClient.url(amlsCreateUrl)
+          .withHeaders(CONTENT_TYPE -> "application/json")
+          .post(payload), 10 seconds
+      )
+
+    def payload(maybeUtr: Option[Utr] = None,
+                supervisory: String = "supervisoryBody",
+                arn: Option[Arn] = None) = {
+      val utr = maybeUtr.getOrElse(Utr(Random.alphanumeric.take(10).mkString("")))
+      val amlsDetails = AmlsDetails(utr, supervisory, "0123456789", LocalDate.now(), arn)
+      Json.toJson(amlsDetails).toString()
+    }
+
+    scenario("user logged in and is an agent should be able to create a new Amls record for the first time") {
+      Given("User is logged in and is an agent")
+      withAffinityGroupAgent
+
+      When("POST /amls/create is called")
+      val response: WSResponse = doRequest(payload())
+
+      Then("201 CREATED is returned")
+      response.status shouldBe 201
+    }
+
+    scenario("User is not logged in") {
+      Given("User is not logged in")
+      isNotLoggedIn
+
+      When("POST /amls/create is called")
+      val response: WSResponse = doRequest(payload())
+
+      Then("401 UnAuthorized is returned")
+      response.status shouldBe 401
+    }
+
+    scenario("update existing amls record but with no ARN should return success") {
+
+      Given("User is logged in and is an agent")
+      withAffinityGroupAgent
+
+      val utr = Some(Utr(Random.alphanumeric.take(10).mkString("")))
+
+      When("POST /amls/create is called")
+      val response: WSResponse = doRequest(payload(utr))
+
+      Then("201 CREATED is returned")
+      response.status shouldBe 201
+
+      When("POST /amls/create is called second time with the same UTR")
+      val newResponse: WSResponse = doRequest(payload(utr,"updated-supervisory"))
+
+      Then("201 CREATED is returned")
+      newResponse.status shouldBe 201
+    }
+
+    scenario("updates to an already existing amls record with an ARN should return error") {
```

Comment:
can we rename to, "create will fail to update existing details for UTR if record updated with ARN"